subsys/bluetooth/controller/ll_sw/pdu.h

```diff
@@ -48,6 +48,14 @@
 				     sizeof(struct pdu_adv_sync_info) + \
 				     TX_PWR_SIZE + \
 				     ACAD_SIZE)
+/* Advertisement channel maximum payload size */
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+#define PDU_AC_PAYLOAD_SIZE_MAX MAX(MIN(CONFIG_BT_CTLR_ADV_DATA_LEN_MAX, \
+					PDU_AC_EXT_PAYLOAD_SIZE_MAX), \
+				    PDU_AC_LEG_PAYLOAD_SIZE_MAX)
```

Comment:
`BT_CTLR_ADV_DATA_LEN_MAX` is maximum length of AD, not PDU payload so even if e.g. `BT_CTLR_ADV_DATA_LEN_MAX 50` we need to support larger payload size to include common ext adv header. I think we should just assume `PDU_AC_EXT_PAYLOAD_SIZE_MAX` or find max overhead for common ext adv given our supported features in code.

Reply:
Max overhead of common ext adv header is 64 bytes, I have updated the PR now. And added a TODO comment for reducing the overhead based on features enabled in controller.